Possible incorrect type annotation in stdlib/urllib/parse.pyi

I have the following sample code, where both mypy and pyright reports an error.

```python
import urllib.parse

query_params = {}
urllib.parse.urlencode(query_params, quote_via=urllib.parse.quote)
```

mypy
```
test.py:3: error: Need type annotation for "query_params" (hint: "query_params: Dict[<type>, <type>] = ...")
test.py:4: error: Value of type variable "AnyStr" of "urlencode" cannot be "Union[bytes, str]"
```

pyright
```
test.py:4:48 - error: Argument of type "Overload[(string: str, safe: _Str = ..., encoding: str | None = ..., errors: str | None = ...) -> str, (string: bytes, safe: _Str = ...) -> str]" cannot be assigned to parameter "quote_via" of type "(_p0: str, _p1: AnyStr@urlencode, _p2: str, _p3: str) -> str" in function "urlencode"
    No overloaded function matches type "(_p0: str, _p1: AnyStr@urlencode, _p2: str, _p3: str) -> str"
      Type "(string: str, safe: _Str = ..., encoding: str | None = ..., errors: str | None = ...) -> str" cannot be assigned to type "(_p0: str, _p1: AnyStr@urlencode, _p2: str, _p3: str) -> str"
        Parameter 2: type "AnyStr@urlencode" cannot be assigned to type "_Str"
      Type "(string: bytes, safe: _Str = ...) -> str" cannot be assigned to type "(_p0: str, _p1: AnyStr@urlencode, _p2: str, _p3: str) -> str"
        Function accepts too many positional parameters; expected 2 but received 4
          Parameter 1: type "str" cannot be assigned to type "bytes"
            "str" is incompatible with "bytes"
          Parameter 2: type "AnyStr@urlencode" cannot be assigned to type "_Str" (reportGeneralTypeIssues)
```

I originally filed a bug in microsoft/pyright#2267. @erictraut has indicated that the issue is likely at [stdlib/urllib/parse.pyi](https://github.com/python/typeshed/blob/master/stdlib/urllib/parse.pyi#L101).

It appears that changing `_Str` to `AnyStr` removes the error.

--- a/stdlib/urllib/parse.pyi
+++ b/stdlib/urllib/parse.pyi
@@ -116,10 +116,10 @@ def urldefrag(url: bytes | None) -> DefragResultBytes: ...
 def urlencode(
     query: Mapping[Any, Any] | Mapping[Any, Sequence[Any]] | Sequence[tuple[Any, Any]] | Sequence[tuple[Any, Sequence[Any]]],
     doseq: bool = ...,
-    safe: AnyStr = ...,
+    safe: _Str = ...,
     encoding: str = ...,
     errors: str = ...,
-    quote_via: Callable[[str, AnyStr, str, str], str] = ...,
+    quote_via: Callable[[AnyStr, _Str, str, str], str] = ...,
 ) -> str: ...
 def urljoin(base: AnyStr, url: AnyStr | None, allow_fragments: bool = ...) -> AnyStr: ...
 @overload

